### PR TITLE
Merge Rust Runner into main

### PR DESCRIPTION
**Pull request by ohjuny**
#1 sets up the `pyo3` environment and replaces the Python-side maximum flow implementation with Rust.
#3 replaces the entire `find_min_cut` implementation with Rust.

The two PRs above was collected into the [`lowtime-rust`](https://github.com/ml-energy/lowtime/tree/lowtime-rust) branch.
This PR merges the branch into main.

### CODE REVIEW

File: .github/workflows/maturin.yml [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+      - uses: actions/download-artifact@v4
+      - name: Generate artifact attestation
+        uses: actions/attest-build-provenance@v1
+        with:
+          subject-path: 'wheels-*/*'
+      - name: Publish to PyPI
+        if: "startsWith(github.ref, 'refs/tags/')"
+        uses: PyO3/maturin-action@v1
+        env:
+          MATURIN_PYPI_TOKEN: ${{ secrets.PYPI_API_TOKEN }}
```
Comment (jaywonchung) [inline .github/workflows/maturin.yml]:
```suggestion
          MATURIN_PYPI_TOKEN: ${{ secrets.PYPI_TOKEN }}
```
Does this replace https://github.com/ml-energy/lowtime/blob/master/.github/workflows/publish_pypi.yaml? If so, we can remove `publish_pypi.yaml`.

File: lowtime/solver.py [hunk 2]
```diff
@@ -243,13 +274,22 @@ def run(self) -> Generator[IterationResult, None, None]:
...
-                s_set, t_set = self.find_min_cut(critical_dag)
+                nodes, edges = self.format_rust_inputs(critical_dag)
+                rust_runner = _lowtime_rs.PhillipsDessouky(
+                    FP_ERROR,
+                    nodes,
+                    critical_dag.graph["source_node"],
+                    critical_dag.graph["sink_node"],
+                    edges,
+                )
+                s_set, t_set = rust_runner.find_min_cut()
```
Comment (jaywonchung) [inline lowtime/solver.py:285]: If you have time, make a screenshot of this diff (replacement of `self.find_min_cut` with these lines is basically all the core code change in lowtime's solver) and put it in your slides. It's extremely cool!

File: pyproject.toml [hunk 3]
```diff
@@ -17,6 +19,8 @@ classifiers = [
     "Programming Language :: Python :: 3.8",
     "Programming Language :: Python :: 3.9",
     "Programming Language :: Python :: 3.10",
+    "Programming Language :: Rust",
+    "Programming Language :: Python :: Implementation :: CPython",
```
Comment (jaywonchung) [inline pyproject.toml]:
```suggestion
```

**Review by jaywonchung — APPROVED** [review]

File: .github/workflows/maturin.yml (hunk 1 above)
Comment (ohjuny) [reply .github/workflows/maturin.yml]: Honestly not sure... is there a way to test this?

Comment (jaywonchung) [reply .github/workflows/maturin.yml]: I'd say it's weird if the maturin release workflow doesn't replace the original release workflow. Let's remove the old one.

File: pyproject.toml (hunk 3 above)
Comment (ohjuny) [reply pyproject.toml]: github gui "commit suggestion" was buggy so made a manual commit for this change.

**Pull request merged**